src/tests/multihost/ad/test_access_control.py

```diff
@@ -12,6 +13,23 @@
 from sssd.testlib.common.exceptions import SSHLoginException
 
 
+def ssh_login(multihost, username):
+    client_hostname = multihost.client[0].sys_hostname
+    client = pexpect_ssh(client_hostname, username, 'Secret123', debug=False)
+    try:
+        client.login()
+    except SSHLoginException:
+        pytest.fail(f'{username} failed to login')
+        return 'false', 'login failed'
```

Comment:
Please put a proper bool there otherwise this is bound to happen:
```
result, message = ssh_login(session_multihost, 'aduser')
if result:
    print("login worked!")
```

Reply:
There are three scenarios involved here. 
1. log in access denied ... this will be checked  (negative scenario)
2. log in succeed (allowed) .... this will be checked
3. log in failed for any other reason

Reply:
Then returning True is misleading and all values should be a string.

Reply:
That looks better, would rather see there bool if the ssh succeeded/failed and the reason/message (denied,failed, whatever) as the second part.
There is also a method auth_from_client : https://github.com/SSSD/sssd/blob/master/src/tests/multihost/sssd/testlib/common/utils.py#L590 that has similar functionality so it might be worth using/extending it to suit Your needs.

Reply:
Ok, please make the strings 'failed', 'denied' and 'success' and add a proper docstring for the function.

Reply:
Added strings